selfservice/strategy/oidc/strategy_registration.go

```diff
@@ -349,13 +362,57 @@ func (s *Strategy) processRegistration(w http.ResponseWriter, r *http.Request, r
 
 	i.SetCredentials(s.ID(), *creds)
 	if err := s.d.RegistrationExecutor().PostRegistrationHook(w, r, identity.CredentialsTypeOIDC, provider.Config().ID, rf, i); err != nil {
+		dup := new(identity.ErrDuplicateCredentials)
+		if provider.Config().AutomaticAccountLinking == AutomaticAccountLinkingReplace && errors.As(err, &dup) {
+			conflictingIdentity, _, err := s.d.IdentityManager().FindConflictingIdentity(r.Context(), i)
+			if err != nil {
+				return nil, s.handleError(w, r, rf, provider.Config().ID, i.Traits, err)
+			}
+			if conflictingIdentity == nil {
+				return nil, s.handleError(w, r, rf, provider.Config().ID, i.Traits, dup)
+			}
```

Comment:
Can this actually ever happen? Returning `nil, "", nil` from `FindConflictingIdentity` can lead to panics if you forget to check the value of the identity for nil, which you don't typically do. E.g. you assume that the values returned are valid if no `err` was returned. Maybe we can return an error from `FindConflictingIdentity` instead?

Reply:
Yeah that sounds sensible. I just extracted the method and the `nil, "", nil` case was when we could not find a conflicting identity. I guess an error is better in this case.

Reply:
Agree, `nil, "", nil` should not be a valid return type of FindConflictingIdentity